Libraries/src/Amazon.Lambda.Annotations/APIGateway/HttpResults.cs

```diff
@@ -0,0 +1,437 @@
+            V1, 
+            /// <summary>
+            /// V2 format for API Gateway Proxy responses. Used for functions defined with HttpApiAttribute with an implicit version of an explicit setting to V2.
+            /// </summary>
+            V2 
+        }
+
+        /// <summary>
+        /// The API Gateway protocol used as the event source. 
+        ///     RestApi -> RestApiAttrbute
+        ///     HttpApi -> HttpApiAttribute
+        /// </summary>
+        public ProtocolFormat Format { get; set; }
+
+        /// <summary>
+        /// The API Gateway protocol version used as the event source. 
+        ///     V1 -> RestApi or HttpApi specifically set as V1
+        ///     V2 -> HttpApi either implicit or explicit set to V2
+        /// </summary>
+        public ProtocolVersion Version { get; set; }
+    }
+
+    /// <summary>
+    /// If this inteface is returned for an API Gateway Lambda function it will serialize itself to the correct JSON format for the 
+    /// configured event source's protocol format and version.
+    /// 
+    /// Users should use the implementation class HttpResults to construct an instance of IHttpResults with the configured, status code, response body and headers.
+    /// </summary>
+    /// <example>
+    /// return HttpResults.Ok("All Good")
+    ///                   .AddHeader("Custom-Header", "FooBar");
+    /// 
+    /// </example>
+    public interface IHttpResults
+    {
+        /// <summary>
+        /// Used by the Lambda Annotations framework to serialize the IHttpResults to the correct JSON response.
+        /// </summary>
+        /// <param name="options"></param>
+        /// <returns></returns>
+        Stream Serialize(HttpResultSerializationOptions options);
+
+        /// <summary>
+        /// Add header to the IHttpResult. The AddHeader method can be called multiple times for the same header to add multi values for a header.
+        /// </summary>
+        /// <param name="name"></param>
+        /// <param name="value"></param>
+        /// <returns>The same isntance to allow fluent call pattern.</returns>
+        IHttpResults AddHeader(string name, string value);
+    }
+
+    /// <summary>
+    /// Implemenation class for IHttpResults. Consumers should use one of the static methods to create the a result with the desired status code.
+    /// </summary>
+    /// <remarks>
+    /// If a response body is provided it is format using the following rules:
+    ///     * For string then returned as is.
+    ///     * For Stream, byte[] or IList&lt;byte&gt; the data is consided binary and base 64 encoded.
+    ///     * Anything other type is serialized to JSON.
+    /// </remarks>
+    /// <example>
+    /// return HttpResults.Ok("All Good")
+    ///                   .AddHeader("Custom-Header", "FooBar");
+    /// 
+    /// </example>
+    public class HttpResults : IHttpResults
+    {
+        private HttpStatusCode _statusCode;
+        private string _body;
```

Comment:
These 2 warnings are misleading because the variables are not used in the stub .NET Standard 2.0 version but are used in the real .NET 6 version.